File: backend/src/main/java/com/matricula/modules/facultad/service/FacultadService.java
package com.matricula.modules.facultad.service;

import com.matricula.modules.facultad.dto.FacultadDTO;

import java.util.List;

public interface FacultadService {
    List<FacultadDTO> findAll();
    FacultadDTO findById(Integer id);
    FacultadDTO create(FacultadDTO dto);
    FacultadDTO update(Integer id, FacultadDTO dto);
    void delete(Integer id);
}

File: frontend/src/services/facultadService.js
import api from './api';

export const facultadService = {
  getAll: () => api.get('/api/facultades'),
  getById: (id) => api.get(`/api/facultades/${id}`),
  create: (facultad) => api.post('/api/facultades', facultad),
  update: (id, facultad) => api.put(`/api/facultades/${id}`, facultad),
  delete: (id) => api.delete(`/api/facultades/${id}`),
};